### PR TITLE
feat(primitives): clean up primitives `proof.rs` file

### LINKED ISSUE #5164: Include `ommers_hash` in `SealedBlock`
### Describe the feature

In the following function:

https://github.com/paradigmxyz/reth/blob/0f254e16a50a6d39543fe59590227a09fd0e21bb/crates/consensus/common/src/validation.rs#L196-L208

the `ommers_hash` is calculated inside the function using the `calculate_ommers_root` method. Why not directly integrate directly this `ommers_hash`field into the `SealedBlock` structure. This could be useful in this function on the one hand but also in other future implementations, what do you think?

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tcoratger**
Should close #5164.

### CODE REVIEW

File: crates/primitives/src/proofs.rs [hunk 1]
```diff
@@ -92,19 +85,15 @@ where
...
-
 /// Calculates the root hash for ommer/uncle headers.
 pub fn calculate_ommers_root(ommers: &[Header]) -> B256 {
     // RLP Encode
     let mut ommers_rlp = Vec::new();
     alloy_rlp::encode_list(ommers, &mut ommers_rlp);
+    // Check if `ommers` list is empty
+    if ommers_rlp.is_empty() {
+        return EMPTY_ROOT_HASH
+    }
```
Comment (mattsse) [inline crates/primitives/src/proofs.rs:96]: this can be moved to the beginning of the function, so we don't even allocate an rlp header

**Review by mattsse — CHANGES_REQUESTED** [review]
smol nit

File: crates/primitives/src/proofs.rs (hunk 1 above)
Comment (tcoratger) [reply crates/primitives/src/proofs.rs]: @mattsse Correct, I misnamed the hash on my first try so it didn't work, now it's ok. Just pushed the update.

**Review by mattsse — APPROVED** [review]

**Comment by mattsse** [comment]
pending failign tests

**Pull request merged**